src/nccl_ofi_scheduler.c

```diff
@@ -35,67 +52,20 @@ void nccl_net_ofi_set_multiplexing_schedule(size_t size, int num_rails,
 
 	schedule->num_xfer_infos = 0;
 
-	if (OFI_UNLIKELY(num_rails == 0)) return;
-
-	max_stripe_size = NCCL_OFI_DIV_CEIL(NCCL_OFI_DIV_CEIL(size, num_rails), align) * align;
+	max_stripe_size = NCCL_OFI_DIV_CEIL(NCCL_OFI_DIV_CEIL(size, num_stripes), align) * align;
 
 	/* Compute stripes and assign to rails */
-	for (int rail_id = 0; rail_id != num_rails && left > 0; ++rail_id) {
+	for (int stripe_idx = 0; stripe_idx == 0 || (stripe_idx < num_stripes && left > 0); ++stripe_idx) {
 		size_t stripe_size = NCCL_OFI_MIN(left, max_stripe_size);
```

Comment:
This needs to have the 128 byte alignment requirement taken into account.

Reply:
Currently we do calculate the max stripe size here where we make sure the stripe size is a multiple of 128
- https://github.com/aws/aws-ofi-nccl/pull/604/files#diff-984de7460dd2ad341a058b8bbe8615963b8550654b7b987b7a8a0ad23ee5fec3R60
`max_stripe_size = NCCL_OFI_DIV_CEIL(NCCL_OFI_DIV_CEIL(size, num_stripes), align) * align;`
Later in - https://github.com/aws/aws-ofi-nccl/pull/604/files#diff-984de7460dd2ad341a058b8bbe8615963b8550654b7b987b7a8a0ad23ee5fec3R60
we calculate stripe_size as below to ensure that the data is LL128 aligned.
size_t stripe_size = NCCL_OFI_MIN(left, max_stripe_size);
I copied over this logic from the existing multiplexer function but please let me know if this is accurate or require any modification.

Reply:
PUt a comment above the line 60 max_stripe_size computation, because that's not at all clear from a fast read of the code.

Reply:
Done